Anlab.Mvc/Services/IOrderService.cs

```diff
@@ -108,28 +107,38 @@ private List<TestItemModel> GetJoined(IList<TestItemPrices> prices, List<TestIte
         /// </summary>
         /// <param name="orderToUpdate"></param>
         /// <returns></returns>
-        public async Task OverwiteOrderWithTestsCompleted(Order orderToUpdate)
+        public async Task<OverwriteOrderResult> OverwiteOrderWithTestsCompleted(Order orderToUpdate)
         {
+            var rtValue = new OverwriteOrderResult();
             if (string.IsNullOrWhiteSpace(orderToUpdate.RequestNum))
             {
                 throw new Exception("RequestNum not populated"); //TODO: Something better
             }
             var testCodes = await _labworksService.GetTestCodesCompletedForOrder(orderToUpdate.RequestNum);
 
-            var testIds = _context.TestItems.Where(a => testCodes.Contains(a.Code)).Select(s => s.Id).ToArray(); //TODO: Currently if the test code doesn't exist in our DB we ignore it, but this could cause issues if the test not found has a $ amount.
+            var testIds = _context.TestItems.Where(a => testCodes.Contains(a.Id)).Select(s => s.Id).ToArray(); 
             var tests = await PopulateSelectedTestsItemModel(testIds);
 
+            if (testCodes.Count != testIds.Length)
+            {
+                //Oh No!!! tests were returned that we don't know about
+                var foundCodes = _context.TestItems.Where(a => testIds.Contains(a.Id)).Select(s => s.Id).Distinct().ToList();
+                var missingCodes = testCodes.Except(foundCodes).ToList();
+                rtValue.Errors.Add(string.Format("Error. Unable to continue. The following codes were not found locally: {0}", string.Join(",", missingCodes)));

```

Comment:
@jSylvestre I don't think this method which should overwrite orders should need to be formatting error strings.  I'd do `rtValue.MissingCodes = testCodes.Except(foundCodes).ToArray()` and then set rtValue.Success = true/false depending on status, and let the calling code decide how to handle missing codes and format the values on the caller side.